pandas/tests/extension/decimal/array.py

```diff
@@ -75,9 +75,19 @@ def isna(self):
         return np.array([x.is_nan() for x in self.values])
 
     def take(self, indexer, allow_fill=True, fill_value=None):
+
         indexer = np.asarray(indexer)
         mask = indexer == -1
 
+        # take on empty array
+        if not len(self):
```

Comment:
same

Reply:
DecimalArray.take does not use `take_1d`, so it's not possible to move it

Reply:
then you need to ask yourself if this is the correct implementation. pandas implementst take_1d for exactly this reason.

Reply:
`take_1d` is not a public function. Do you want to expose it publicly?